src/highlight.rs

```diff
@@ -44,6 +45,14 @@ pub static DATA: LazyLock<Data> = LazyLock::new(|| {
     }
 });
 
+static LINK_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+    Regex::new(r#"https?://[^ "]+"#).unwrap()
```

Comment:
Maybe one should also exclude tabs, and to be sure newline, so `https?://[^ \t\n\r"]+`.

But then what about links embedded in other ways:

```
``http://example.org?foo=`bar```
[http://example.org]
(http://example.org)
```

Probably overkill but:
```regex
(?:[^"'`\[\(<])?(https?://[^ \n\r\t"'`\]\)>]+)|"(https?://[^ \n\r\t"]+)"|```(https?://[^ \n\r\t]+)```|``(https?://[^ \n\r\t]+)``|`(https?://[^ \n\r\t`]+)`|\((https?://[^ \n\r\t`]+)\)|\[(https?://[^ \n\r\t`]+)\]
```